x-pack/plugins/canvas/public/lib/clone_subgraphs.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { arrayToMap } from './aeroelastic/functional';
+import { getId } from './get_id';
+
+export const cloneSubgraphs = nodes => {
+  const idMap = arrayToMap(nodes.map(n => n.id));
+  // We simultaneously provide unique id values for all elements (across all pages)
+  // AND ensure that parent-child relationships are retained (via matching id values within page)
+  Object.entries(idMap).forEach(([key]) => (idMap[key] = getId(key.split('-')[0]))); // new group names to which we can map
```

Comment:
Since you're discarding the value in the entry, wouldn't `Object.keys(idMap).forEach(key => ...` do the same thing here?

Reply:
Haha good spot, I overlooked this one, apparently ES2015 zealot here 😄 Changing it now

Reply:
Done, will push; an initial version used the value too and it got stuck :-)

Reply:
I only dug in because I haven't used `Object.entries` before and had to look it up ;)